vignettes/upscaling.Rmd

```diff
@@ -1,43 +1,102 @@
+Let's import our site by locations object.
+
 
-```{r upscale-richness}
-## Upscale to grid ----
-ras_richness <- fb_aggregate_site_data(site_locations = site_locations, 
-                                       site_data      = rich_env, 
-                                       agg_grid       = tavg)
+```{r import-site-locations}
+data("site_locations")
 
-ras_richness
+site_locations
 ```
 
-```{r map-richness, dev = 'png'}
-fb_map_raster(ras_richness[[1]])
+These sites are a collection of regular spatial polygons at a resolution of 0.5°.
+
+For each site, we want to compute the species richness. 
+
+
+```{r compute-richness}
+# Import site x species data ----
+data("site_species")
+
+# Compute species richness ----
+species_richness <- fb_count_species_by_site(site_species)
+
+head(species_richness)
 ```
 
-```{r map-richness-custom, dev = 'png'}
-fb_map_raster(ras_richness[[1]]) + 
-  scale_fill_distiller("Number of species", palette = "Spectral") +
-  theme(legend.position = "bottom") + 
-  ggtitle("Trees species richness in Pennsylvania")
+Before going any further let's map these original values with the function
+`fb_map_site_data()`.
+
+
+```{r map-original-richness, dev='png'}
+fb_map_site_data(site_locations, species_richness, "n_species")
 ```
 
+Now, let's say that our next analyses require to work at a coarser resolution. We need to
+aggregate site data on a new spatial grid (object `SpatRaster` from the 
+[`terra`](http://cran.r-project.org/package=terra) package). Let's import this coarser raster.
+
+
+```{r import-grid}
+# Import study area grid ----
+grd <- system.file("extdata", "grid_area.tif", package = "funbiogeo")
+grd <- terra::rast(grd)
+
+grd
+```
+
+We will aggregate the site data (resolution of 0.5°) to this raster (resolution of 0.83°) 
+with the function `fb_aggregate_site_data()`. This function requires the following arguments:
+
+- `site_locations`: the site x locations object
+- `site_data`: a `matrix` or `data.frame` containing values per sites to aggregate on the provided grid `agg_grid`. Can have one or several columns (variables to aggregate). The first column must contain sites names as provided in the example dataset `site_locations`
+- `agg_grid`: a `SpatRaster` object (package `terra`). A raster of one single layer, that defines the grid along which to aggregate
+- `fun`: the function used to aggregate sites values when there are multiple sites in one cell
+
+Let's aggregate our species richness values on this grid.
+
+
+```{r upscaled-richness}
+# Upscale to grid ----
+upscaled_richness <- fb_aggregate_site_data(
+  site_locations = site_locations,
+  site_data      = species_richness[ , 1:2],
+  agg_grid       = grd,
+  fun            = max
+)
+
+upscaled_richness
+```
+
+The result of this function is a `SpatRaster` where the cells contain the aggregated values of the species richness.
+
+
+```{r map-upscaled-richness, dev='png'}
+fb_map_raster(upscaled_richness)
+```
```

Comment:
I would add an example of aggregating site-species matrix here, as well as CWM (because we're in a functional perspective), but I can write this.

Reply:
Good idea. If you can add this example it could be great. Maybe w/ `fundiversity`/`mFD`?

Reply:
Thanks